backend/editor/api.py

```diff
@@ -35,4 +57,214 @@ async def pong(response: Response):
+    result = session.run(query, {"id": entry})
+    oneEntry = [record for record in result]
+    return oneEntry
+
+@app.get("/entriesfull")
+async def findAllEntries(response: Response):
+    """
+    Get all entries within taxonomy
+    """
+    query = """
+        MATCH (n:ENTRY) RETURN n
+    """
+    result = session.run(query)
+    allEntries = [record for record in result]
+    return allEntries
+
+@app.get("/synonym/{synonym}")
+async def findOneSynonym(response: Response, synonym: str):
+    """
+    Get synonym corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:SYNONYMS) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": synonym})
+    oneSynonym = [record for record in result]
+    return oneSynonym
+
+@app.get("/synonymsfull")
+async def findAllSynonyms(response: Response):
+    """
+    Get all synonyms within taxonomy
+    """
+    query = """
+        MATCH (n:SYNONYMS) RETURN n
+    """
+    result = session.run(query)
+    allSyononyms = [record for record in result]
+    return allSyononyms
+
+@app.get("/stopword/{stopword}")
+async def findOneStopword(response: Response, stopword: str):
+    """
+    Get stopword corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:STOPWORDS) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": stopword})
+    oneStopword = [record for record in result]
+    return oneStopword
+
+@app.get("/stopwordsfull")
+async def findAllStopwords(response: Response):
+    """
+    Get all stopwords within taxonomy
+    """
+    query = """
+        MATCH (n:STOPWORDS) RETURN n
+    """
+    result = session.run(query)
+    allStopwords = [record for record in result]
+    return allStopwords
+
+
+@app.get("/header")
+async def findHeader(response: Response):
+    """
+    Get __header__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__header__"
+        RETURN n
+    """
+    result = session.run(query)
+    header = [record for record in result]
+    return header
+
+@app.get("/footer")
+async def findFooter(response: Response):
+    """
+    Get __footer__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__footer__"
+        RETURN n
+    """
+    result = session.run(query)
+    footer = [record for record in result]
+    return footer
+
+# Put methods 
+
+@app.put("/edit/entry/{entry}")
+async def editEntry(request: Request, entry: str):
+    """
+    Editing an entry in a taxonomy.
+    New key-value pairs can be added, old key-value pairs can be updated.
+    URL will be of format '/edit/<id>'
+    """
+    incomingData = await request.json()
+    result = None
+    for key in incomingData.keys():
+        query = f"""
+            MATCH (n:ENTRY) WHERE n.id = $id
+            SET n.{key} = $value
+            RETURN n
+        """
+        result = session.run(query, {"id": entry, "value": incomingData[key]})

```

Comment:
You should do it all in one query, this will be much more efficient.

You just have to build the "SET" lines and concatenate them. For each line instead of $value you can use the key name `f"SET .... ${key}"` and pass the whole dict to "run", adding the "id".

Reply:
see how I did this in sample/load.py

Reply:
@alexgarel I'm not able to figure out where exactly you've written the "concatenation" step in load.py. Would you be able to point that out to me?

Reply:
I built the "SET" lines by concatenation and used it as a query.